jsui/fluid.plotter.js

```diff
@@ -241,26 +241,28 @@ function setcategories(name) {
 
 function constructColorScheme() {
 	if (labelDict) {
-		const data = labelDict.get('data');
-		const keys = data.getkeys();
-		var uniques = new Array();
+		var labelDictAsJson = JSON.parse(labelDict.stringify());
+		var data = labelDictAsJson.data;
+		var keys = Object.keys(data)
+		var uniques = []
 
 		// How many unique labels are there?
 		keys.forEach(function(key) {
-			var label = data.get(key);
+			var label = data[key]
 			if (uniques.indexOf(label) == -1) {
-				uniques.push(label)
+				uniques.push(label[0])
 			}
 		})
-		
-		colorMap = {};
-		var scheme = strChunk(_colorscheme, 6);
+		uniques = uniques.removeDuplicates()
```

Comment:
why do we need to run this here? the keys.forEach is not supposed to add anything that is found, right?

Reply:
In theory you should not need it, but it makes for a good guard against the original problem emerging.

It is very cheap to run so I would opt to just keep it, even though it is redundant.

Reply:
Actually yeah looking at that its very meaningless to put this into an array prototype function.

I did update the check to actually work, which is why I think I needed the removal of duplicates: https://github.com/flucoma/flucoma-max/pull/412/commits/a2f90e0337fa8274ab7683c59f5374e7fd5e3e70